internal/home_server_url.go

```diff
@@ -0,0 +1,25 @@
+package internal
+
+import "strings"
+
+type HomeServerUrl struct {
+	HttpOrUnixStr string
+}
+
+func (u HomeServerUrl) IsUnixSocket() bool {
```

Comment:
This isn't what I was getting at before. It would be quite enough to just have:
```go
func IsUnixSocket(u string) bool {
    return strings.HasPrefix(u, "/")
}
```
This is better than before because then the implementation of prefix checking is only in a single place, so if we want to swap to something better-than-prefix-checking we just need to edit one place, and it reads more nicely and self-documents:
```go
if internal.IsUnixSocket(destHomeserverUrl) {
    // set up unix transport
}
```

We don't need the boilerplate struct and getters here, as they don't add any value. They don't add value because the caller still needs to care about the response from this function to determine if the call to `GetUnixSocket()` or `GetBaseUrl()` would make any sense, so the caller's code would look a lot like the above, just with a struct definition instead of a simple static function. If we were going to go down the struct road, it would be better to have a way to map from a `HttpOrUnixStr` to an `http.Transport`, which could be called without caring about whether this is a unix or HTTP string.